python/google/protobuf/message_factory.py

```diff
@@ -122,6 +115,32 @@ def GetMessages(self, files):
         extended_class.RegisterExtension(extension)
     return result
 
+  def ClearCache(self):
+    """Clear the MessageFactory's cache of which classes it knows about. This
+    is typically only something you need to do if you've been programmatically
+    importing _pb2.py files and want to make sure the factory understands them
+    correctly.
+    """
+    self._classes = {}
+
+  def _GetPrototypeClass(self, descriptor):
+    try:
```

Comment:
No need to add  _GetPrototypeClass(self, descriptor), just move the try/except to GetPrototype(self, descriptor) should be enough

Reply:
Done.